### PR TITLE
Fix DW pushing same request multiple times.

### PR DESCRIPTION
**Pull request by lljrsr**
DW was pushing multiple requests at the same time. This is related to issue #87 . However I did not test if redirects are working yet.

### CODE REVIEW

**Comment by sibiryakov** [comment]
I would start with investigating how these URLs passed the SW state cache. Scheduling of multiple URLs shouldn't happen.

**Comment by lljrsr** [comment]
I do not think that SW are pushing multiple URLs. I think that this is a bug in HBase Backend where each row key is added multiple times to the new batch.

File: frontera/contrib/backends/hbase.py [hunk 1]
```diff
@@ -223,17 +223,19 @@ def get_next_requests(self, max_n_requests, partition_id, **kwargs):
...
                 for rk, _ in meta_map[fprint]:
-                    trash_can.add(rk)
-                    for rk_fprint in fprint_map[rk]:
-                        _, item = meta_map[rk_fprint][0]
-                        _, _, url, score = item
-                        results.append(Request(url, meta={
-                            'fingerprint': hexlify(rk_fprint),
-                            'score': score,
-                        }))
+                    if rk not in trash_can:
```
Comment (sibiryakov) [inline frontera/contrib/backends/hbase.py]: What if we rewrite it:
```python
if rk in trash_can:
    continue
```
It will prevent growing of indent.

**Comment by sibiryakov** [comment]
You were right actually. It generates plenty of duplicates, perhaps I left this error after refactoring to metadata/queue/states model.

**Comment by sibiryakov** [comment]
All items stored at some RK, were collected to ``results`` list len(rk) times...

**Comment by lljrsr** [comment]
Awesome :) . So let's merge this with master. I was a bit unsure about renaming items to requests, since there are so many different mentionings of items in the whole project. Maybe that is something you could do in a different branch to differentiate between requests and items.

**Comment by sibiryakov** [comment]
Your renaming was a right thing here. Because *items* related to Items Pipeline in Scrapy, which is absolutely different thing.

**Comment by sibiryakov** [comment]
Could you please rebase your branch, merging everything to one commit and putting clear commit message? It could be done with
```bash
$git rebase -i HEAD~4 (you have 4 commits in your branch)
$git log (check if it looks ok, check the code)
$git push -f
```

**Comment by lljrsr** [comment]
It looks like everything worked fine :) .

**Comment by sibiryakov** [comment]
Great :beers: !

**Pull request merged**